frontend/src/components/doctors/doctors/DoctorDistance.tsx

```diff
@@ -0,0 +1,54 @@
+import {
+    DoctorLocation,
+    getDoctorLocationDistance,
+} from "../../../types/doctors/doctor";
+import { Location } from "../../../utils/googleMaps/useGoogleMaps";
+import { DistanceUnit } from "../../utils/DistanceUnit";
+import Icon from "../../utils/Icon";
+
+interface DoctorDistanceProps {
+    location: DoctorLocation | null | undefined;
+    locationForDistanceCalculation?: Location;
+    distanceUnit?: DistanceUnit;
+}
+
+function DoctorDistance({
+    location,
+    locationForDistanceCalculation,
+    distanceUnit,
+}: DoctorDistanceProps) {
+    const distance =
+        locationForDistanceCalculation &&
+        location &&
+        getDoctorLocationDistance(
+            location,
+            locationForDistanceCalculation,
+            distanceUnit
+        );
+
+    return (
+        <>
+            {distance && distance !== Infinity && (
+                <a
+                    href={`http://maps.google.com/maps/dir/?api=1&dir_action=navigate&destination=${
+                        location.address || ""
```

Comment:
If location.address is undefined, we shouldn't get to this code... A link with "" doesn't make sense.